### ISSUE TITLE
pylint: add an option to control the exit code

### ISSUE THREAD

**Issue by dgutson** [pylint-dev/pylint#8672]
### Current problem

I need to determine whether pylint failed (eg crashed, invalid arguments, file not found) or ir found issues.

Usually in *nix systems, exit codes greater than 127 mean that either the process received a signal (kill, segfault) or there was another kind of error.

However, pylint returns the amount of issues found (truncated).

### Desired solution

I propose to add a flag to tell pylint to use the exit code for reporting errors and letting the OS report signals.

For example, `pylint --exit-code-error x.py`

We could have an enumeration of errors (eg 2 means file not found), or just leave it alone for the OS.

**Comment by Pierre-Sassoulas** [comment]
The pylint exit code is byte encoded depending on what was raised:

    Fatal (1)
    Error (2)
    Warning (4)
    Convention (8)
    Refactor (16)
    Information (NA)

See https://pylint.readthedocs.io/en/latest/user_guide/messages/index.html

It means what you want is already possible, any output where ``exit_code %1 == 0`` exited in error.

**Comment by Franco0700** [comment]
But I'm having this
```
$ pylint folder/*.py                                                     
************* Module internal-module1
folder/file.py:1:0: C0114: Missing module docstring (missing-module-docstring)
folder/file.py:5:0: E0401: Unable to import 'internal-module' (import-error)
folder/file.py:20:41: C0103: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
************* Module internal-module2
folder/file.py:41:0: C0301: Line too long (103/100) (line-too-long)
folder/file.py:1:0: C0114: Missing module docstring (missing-module-docstring)
... (a lot more issues)
$ echo $?
30
```

**Comment by dgutson** [comment]
@Franco0700 I think I understand: 30 is not the amount of errors, but binary-encoded according to those values.

We should check it is not fatal and it is not greater than 126.
